docs/samples/Microsoft.ML.Samples/Dynamic/SimpleDataViewImplementation.cs

```diff
@@ -0,0 +1,258 @@
+                ReadOnlyMemory<char> textValue = default;
+                VBuffer<ReadOnlyMemory<char>> tokensValue = default;
+
+                var textGetter = cursor.GetGetter<ReadOnlyMemory<char>>(textColumn);
+                var tokensGetter = cursor.GetGetter<VBuffer<ReadOnlyMemory<char>>>(tokensColumn);
+
+                while (cursor.MoveNext())
+                {
+                    textGetter(ref textValue);
+                    tokensGetter(ref tokensValue);
+
+                    Console.WriteLine($"{textValue} => {string.Join(", ", tokensValue.DenseValues())}");
+                }
+
+                // The output to console is this:
+
+                // Hello my friend. => Hello, my, friend.
+                // Stay awhile and listen. => Stay, awhile, and, listen.
+                // Masterfully done hero! => Masterfully, done, hero!
+
+                // Note that it may be interesting to set a breakpoint on the Console.WriteLine, and explore
+                // what is going on with the cursor, and the buffers. In particular, on the third iteration,
+                // while `tokensValue` is logically presented as a three element array, internally you will
+                // see that the arrays internal to that structure have (at least) four items, specifically:
+                // `Masterfully`, `done`, `hero!`, `listen.`. In this way we see a simple example of the details
+                // of how buffer sharing from one iteration to the next actually works.
+            }
+        }
+
+        private sealed class InputObject
+        {
+            public bool Label { get; }
+            public string Text { get; }
+
+            public InputObject(bool label, string text)
+            {
+                Label = label;
+                Text = text;
+            }
+        }
+
+        /// <summary>
+        /// This is an implementation of <see cref="IDataView"/> that wraps an <see cref="IEnumerable{T}"/>
+        /// of the above <see cref="InputObject"/>. Note that normally under these circumstances, the first
+        /// recommendation would be to use a convenience like
+        /// <see cref="DataOperationsCatalog.LoadFromEnumerable{TRow}(IEnumerable{TRow}, SchemaDefinition)"/>
+        /// or something like that, rather than implementing <see cref="IDataView"/> outright. However, sometimes when
+        /// code generation is impossible on some situations, like Unity or other similar platforms, implementing
+        /// something even closely resembling this may become necessary.
+        ///
+        /// This implementation of <see cref="IDataView"/> is, in fact, much simpler than practically anything you would
+        /// find in the ML.NET codebase. In this case we have a completely fixed schema (the two fields of
+        /// <see cref="InputObject"/>), with fixed types. However, for the purpose of the sample, we keep things as
+        /// simple as possible, but without being simplistic, I hope.
```

Comment:
Maybe we can remove this sentence? I am not sure what it's objective is. You already say above that this is simplification of what normally happens in ML.NET.

Reply:
Well... I am saying it's simpler than anything you'll find in the codebase, but I think I'm trying to say that's for the sake of the example. I guess I could compress it by mentioning in the first sentence that it is didactic, then removing this sentence... that's shorter at least.